optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -526,36 +526,380 @@ private static Runnable innerCountDistinctLong(CountDistinctLongResultContainer
     // min
     // ************************************************************************
 
+    /**
+     * Returns a collector that finds a minimum value in a group of {@link Comparable} elements.
+     * <p>
+     *     It is expected that the {@link Comparable}'s {@link Comparable#compareTo(Object)} is consistent with
+     *     {@link #equals(Object)}.
+     *     In other words, if two elements compare to equal, any of them may be returned by the collector as they are
+     *     considered to be {@link #equals(Object)}.
+     * <p>
+     *     Example: Assume the following elements of type Person: Ann(age = 20), Beth(age = 25), Cathy(age = 30),
+     *     David(age = 25), Eric(age = 20).
+     *     Further assume that the Person type compares by age.
+     *     Such collector returns either Ann or Eric as the minimum element.
+     *
+     * @param <A> type of the matched fact
+     * @return never null
+     */
+    public static <A extends Comparable<A>> UniConstraintCollector<A, ?, A> min() {
+        return min(Function.identity(), Comparable::compareTo);
+    }
+
+    /**
+     * Returns a collector that finds a minimum value in a group of {@link Comparable} elements.
+     * <p>
+     *     It is expected that the {@link Comparable}'s {@link Comparable#compareTo(Object)} is consistent with
+     *     {@link #equals(Object)}.
+     *     In other words, if two elements compare to equal, any of them may be returned by the collector as they are
+     *     considered to be {@link #equals(Object)}.
+     * <p>
+     *     Example: Assume the following elements of type Person: Ann(age = 20), Beth(age = 25), Cathy(age = 30),
+     *     David(age = 25), Eric(age = 20).
+     *     Using Person::getAge as the groupValueMapping, the resulting collector returns 20, as that is the minimum
+     *     age of all the elements.
+     *
+     * @param <A> type of the matched fact
+     * @param <Mapped> type of the result
+     * @param groupValueMapping never null, maps facts from the matched type to the result type
+     * @return never null
+     */
+    public static <A, Mapped extends Comparable<Mapped>> UniConstraintCollector<A, ?, Mapped> min(
+            Function<A, Mapped> groupValueMapping) {
```

Comment:
Soft suggestion: I'd order this method (min(Function)) before min(Comparator), because we want to stimulate users to use this one (I know, it doesn't change the order in javadocs or code completion, but people do look at the sources too and for us it's a signal too that we consider this one more important).